File: 61-rotate-list/61-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    
    public ListNode rotate(ListNode head)
    {
        ListNode t=head;
        ListNode p=null;
        while(t.next!=null)
        {
            p=t;
            t=t.next;
        }
        if(p!=null)
        {
             p.next=null;
        }
        t.next=head;
        return t;
        
    }
    
    public ListNode rotateRight(ListNode head, int k) {
        
        if(head==null||head.next==null)
        {
            return head;
        }
        int siz=0;
        ListNode p=head;
        while(p!=null)
        {
            p=p.next;
            siz++;
        }
        if(k>siz)
        {
            k=k%siz;
        }
        for(int i=0;i<k;i++)
        {
            head=rotate(head);
        }
        return head;
    }
}

File: 61-rotate-list/61-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        
        int siz=0;
        ListNode* temp,*lastNode,*newHead;
        int i,j;
        if(head==NULL||head->next==NULL)
        {
            return head;
        }
        temp=head;
        while(temp!=NULL)
        {
            siz++;
            lastNode=temp;
            temp=temp->next;
        }
        if(k>=siz)
        {
            k=k%siz;
        }
        if(k==0)
        {
            return head;
        }
        temp=head;
        i=1;
        while(temp!=NULL&&i<siz-k)
        {
           temp=temp->next;
           i++;
        }
        newHead=temp->next;
        temp->next=NULL;
        lastNode->next=head;
        return newHead;
    }
};